src/connection.ts

```diff
@@ -4567,13 +4553,25 @@ export class Connection {
       feeCalculator: FeeCalculator;
     }>
   > {
-    const args = this._buildArgs([], commitment);
-    const unsafeRes = await this._rpcRequest('getRecentBlockhash', args);
-    const res = create(unsafeRes, GetRecentBlockhashAndContextRpcResult);
-    if ('error' in res) {
-      throw new SolanaJSONRPCError(res.error, 'failed to get recent blockhash');
-    }
-    return res.result;
+    const {
+      context,
+      value: {blockhash},
+    } = await this.getLatestBlockhashAndContext(commitment);
+    return {
+      context,
+      value: {
+        blockhash,
+        feeCalculator: {
+          get lamportsPerSignature(): number {
+            throw new Error(
+              'The capability to fetch `lamportsPerSignature` using the `getRecentBlockhash` API is ' +
+                'no longer offered by the network. Use the `getFeeForMessage` API to obtain the fee ' +
+                'for a given message.',
+            );
+          },
```

Comment:
You know what's annoying? This works just fine with APIs like `console.log`, but if you try to pass this to `JSON.stringify()` it will trigger a read and blow up, which could probably really screw up some error logging products.

```ts
const a = {stuff: 'things', get foo() { throw new Error('o no'); }}
console.log(a); // OK
JSON.stringify(a); // BOOM
```

Add this:

```ts
  get lamportsPerSignature(): number { throw new Error(/* ... */); },
  toJSON() {
    const safe = {};
    Object.keys(this)
      .filter((k) => k !== "lamportsPerSignature")
      .forEach((k) => {
        safe[k] = this[k];
      });
    return safe;
  },
```

…and then maybe a test for that.

Reply:
Sure, but we are already doing this in a few areas in `Connection` already. Do you want all of those to get this treatment?

Reply:
Doing what? Throwing? Every response that would nuke JSON.stringify should probably get protected against doing so, yeah.

Reply:
Ohh I misunderstood the exact problem. This is the only spot where we do this on a field.

I can add this, but TypeScript will barf on it unless I also add `toJSON?: object` to the `FeeCalculator` interface. Additionally, since `lamportsPerSignature` is the only field, we can just make this return `{}` can't we?

Reply:
Add it with `Object.defineProperty()`.

Reply:
Haha actually you can trick TypeScript by just declaring it before returning it. 🫠

Reply:
“Structural typing.”